lib/src/rules/join_return_with_assignment.dart

```diff
@@ -107,17 +106,27 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
     final secondLastStatement = statements[length - 2];
     final lastStatement = statements.last;
-    final secondLastElement =
-        _getElementFromAssignmentStatement(secondLastStatement);
-    final lastElement = _getElementFromReturnStatement(lastStatement);
-    Element thirdLastElement;
+    final secondLastExpression =
+        _getExpressionFromAssignmentStatement(secondLastStatement);
+    final lastExpression = _getExpressionFromReturnStatement(lastStatement);
+
+    // In this case, the last statement was not a return statement with a
+    // simple target.
+    if (lastExpression == null) {
+      return;
+    }
+
+    Expression thirdLastExpression;
     if (length >= 3) {
       final thirdLastStatement = statements[length - 3];
-      thirdLastElement = _getElementFromAssignmentStatement(thirdLastStatement);
+      thirdLastExpression =
+          _getExpressionFromAssignmentStatement(thirdLastStatement);
     }
-    if (lastElement != null &&
-        secondLastElement != thirdLastElement &&
-        lastElement == secondLastElement) {
+    if (lastExpression != null &&
```

Comment:
Unless I'm missing something, `lastExpression` cannot be `null` at this point because of the `if` on line 115. We should remove this condition.